library/Zend/Db/Sql/Predicate/Expression.php

```diff
@@ -19,7 +19,7 @@ class Expression extends BaseExpression implements PredicateInterface
      * @param string $expression
      * @param int|float|bool|string|array $valueParameter
      */
-    public function __construct($expression = null, $valueParameter = null /*[, $valueParameter, ... ]*/)
+    public function __construct($expression = null, $valueParameter = array() /*[, $valueParameter, ... ]*/)
```

Comment:
`$foo = is_array($valueParameter) ? $valueParameter : array_slice(func_get_args(), 1)`
is identical to
```
if (is_array($valueParameter)) {
    $foo = $valueParameter;
} else {
    $foo = array_slice(func_get_args(), 1);
}
```
So, with `$valueParameter == null` i.e. `!is_array($valueParameter)` the `array_slice()` part is rendered :-?